### DISCUSSION TITLE
How to deal with "unexpected end of input"

### DISCUSSION THREAD

**Discussion by JustSimplyKyle** [fzyzcjy/flutter_rust_bridge#1377] [Q&A]
https://github.com/RPMTW/era-connect/commit/d6a2f6ca04db929e7707ddc0314593278f46b879#diff-d48e8ae5ebc890363e450b17ad33bd51ce3a9a08525989aa850dd0ed9a0449d7
Before this change, it compiled perfectly! But now, it errors out with this.
```bash
2023/09/26 18:07:46 [ERROR] panicked at 'called `Result::unwrap()` on an `Err` value: Error("unexpected end of input, expected parentheses")', /home/kyle/.cargo/registry/src/index.crates.io-6f17d22bba15001f/flutter_rust_bridge_codegen-1.82.1/src/parser/markers.rs:16:22
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error("unexpected end of input, expected parentheses")', /home/kyle/.cargo/registry/src/index.crates.io-6f17d22bba15001f/flutter_rust_bridge_codegen-1.82.1/src/parser/markers.rs:16:22
```
Some backtrace
![image](https://github.com/fzyzcjy/flutter_rust_bridge/assets/68589851/29412461-2b81-4059-8661-ba727b42fc57)

I _really_ fail to see where the issue is, maybe the panic can tell at least the location where the issue happens?

**Comment by fzyzcjy** [comment]
It may be easiest to PR to modify the marker.rs a little bit (`src/parser/markers.rs:16:22` as mentioned in the output), such that when it errors, we can see what part of code input triggers the error.

Alternatively, we can make a minimal reproducible sample, for example by bisecting your api.rs.

**Reply by JustSimplyKyle** [reply]
Got it! I'll have a try whenever I have time
Thanks for the direction hint

**Reply by fzyzcjy** [reply]
You are welcome! Looking forward to further information
